example/src/main/java/com/graphhopper/example/RoutingExampleTC.java

```diff
@@ -8,55 +8,81 @@
         GraphHopper hopper = createGraphHopperInstance(relDir + "core/files/andorra.osm.pbf");
-        routing(hopper);
+        routeWithTurnCosts(hopper);
+        routeWithTurnCostsAndCurbsides(hopper);
+        routeWithTurnCostsAndOtherUTurnCosts(hopper);
     }
 
-    // see RoutingExample for more details
-    public static void routing(GraphHopper hopper) {
-        GHRequest req = new GHRequest(42.50822, 1.533966, 42.506899,1.525372).
+    public static void routeWithTurnCosts(GraphHopper hopper) {
+        GHRequest req = new GHRequest(42.50822, 1.533966, 42.506899, 1.525372).
+                setProfile("car");
+        route(hopper, req, 1038, 63_000);
+    }
+
+    public static void routeWithTurnCostsAndCurbsides(GraphHopper hopper) {
+        GHRequest req = new GHRequest(42.50822, 1.533966, 42.506899, 1.525372).
                 setCurbsides(Arrays.asList(CURBSIDE_ANY, CURBSIDE_RIGHT)).
                 setProfile("car");
-        GHResponse rsp = hopper.route(req);
+        route(hopper, req, 1730, 112_000);
+    }
 
+    public static void routeWithTurnCostsAndOtherUTurnCosts(GraphHopper hopper) {
+        GHRequest req = new GHRequest(42.50822, 1.533966, 42.506899, 1.525372).
+                setCurbsides(Arrays.asList(CURBSIDE_ANY, CURBSIDE_RIGHT)).
+                putHint(Parameters.CH.DISABLE, true).
+                setProfile("car");
+        route(hopper, req.putHint(Parameters.Routing.U_TURN_COSTS, 10), 1370, 100_000);
+        route(hopper, req.putHint(Parameters.Routing.U_TURN_COSTS, 15), 1370, 105_000);
+    }
+
+    private static void route(GraphHopper hopper, GHRequest req, int expectedDistance, int expectedTime) {
+        GHResponse rsp = hopper.route(req);
         // handle errors
         if (rsp.hasErrors())
             // if you get: Impossible curbside constraint: 'curbside=right'
             // you either specify 'curbside=any' or Parameters.Routing.FORCE_CURBSIDE=false to ignore this situation
             throw new RuntimeException(rsp.getErrors().toString());
-
         ResponsePath path = rsp.getBest();
-        assert Helper.round(path.getDistance(), -2) == 1700;
+        assert Math.abs(expectedDistance - path.getDistance()) < 1 : "unexpected distance : " + path.getDistance() + " vs. " + expectedDistance;
+        assert Math.abs(expectedTime - path.getTime()) < 1000 : "unexpected time : " + path.getTime() + " vs. " + expectedTime;
     }
 
     // see RoutingExample for more details
     static GraphHopper createGraphHopperInstance(String ghLoc) {
         GraphHopper hopper = new GraphHopperOSM().forServer();
         hopper.setDataReaderFile(ghLoc);
         hopper.setGraphHopperLocation("target/routing-tc-graph-cache");
-        // to enable turn restriction and curbside support ensure that FlagEncoder and profile supports turn costs
-        FlagEncoder car = new CarFlagEncoder(new PMap().
-                putObject("u_turn_costs", 3).putObject("max_turn_costs", 1));
```

Comment:
setting u_turn_costs on the encoder has no effect (?!). also I think for the example its better to use `|turn_costs=true` instead of `max_turn_costs=1` since its easier to understand (we either make or do not make use of the OSM turn restrictions)

Reply:
> setting u_turn_costs on the encoder has no effect (?!)

Ups. Now I start confusing profiles with encoders already myself :D 

> also I think for the example its better to use |turn_costs=true

Ok, sure.